File: tests/fixtures/sample.ts
export function processLogin(username: string, password: string): boolean {
  return username.length > 0 && password.length > 0
}

export function validateToken(token: string): boolean {
  return token.startsWith('Bearer ')
}

export class AuthError extends Error {
  constructor(message: string) {
    super(message)
    this.name = 'AuthError'
  }
}

export interface AuthConfig {
  tokenExpiry: number
  maxRetries: number
}

export type AuthResult = { success: boolean; token?: string }

// internal — should NOT be extracted
function hashPassword(password: string): string {
  return password
}

File: tests/fixtures/sample.py
def process_login(username: str, password: str) -> bool:
    return len(username) > 0 and len(password) > 0


def validate_token(token: str) -> bool:
    return token.startswith("Bearer ")


class AuthError(Exception):
    def __init__(self, message: str):
        super().__init__(message)
        self.name = "AuthError"

    def to_dict(self) -> dict:
        return {"error": str(self)}


# internal — should NOT be extracted
def _hash_password(password: str) -> str:
    return password
